### PR TITLE
Feature/ins 244 listing locations card

### PR DESCRIPTION
**Pull request by vierrra**

### CODE REVIEW

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 1]
```diff
@@ -0,0 +1,282 @@
+//
+//  ListingPlacesCell.swift
+//  StantUiIosLibrary
+//
+//  Created by Renato Vieira on 6/11/20.
+//
+
+import UIKit
+
+public class ListingPlacesCell: UITableViewCell {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Change this name to PlaceCell. It's more simple and direct.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    
+    public static let cellHeight:    CGFloat = 56
+    public var delegate:             ListingPlacesCellDidSelectDelegate?
+    public var bar:                  UIView?
+    public var locationImageView:    UIView?
+    public var locationImage:        UIImageView?
+    public var placeTitleLabel:      UILabel?
+    public var subPlacesLabel:       UILabel?
+    public var percentagePlacelabel: UILabel?
+    public var fispIconView:         UIView?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Fisp is a portuguese acronym. Change it to siffIconView.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    public var fispIconImage:        UIImageView?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Change it to siffIconImage

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+        super.layoutSubviews()
+
+        let padding = UIEdgeInsets(top:    0,
+                                   left:   0,
+                                   bottom: 2,
+                                   right:  4)
+        bounds      = bounds.inset(by: padding)
+    }
+    
+    public func configureViewFor(delegate: ListingPlacesCellDidSelectDelegate, status: PlaceStatusEnum, placeTitle: String, quantitySubPlaces: Int, percentage: Float, hasFips: Bool) {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Break the parameters and indent them.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+//
+
+import UIKit
+
+public class ListingPlacesCell: UITableViewCell {
+    
+    public static let cellHeight:    CGFloat = 56
+    public var delegate:             ListingPlacesCellDidSelectDelegate?
+    public var bar:                  UIView?
+    public var locationImageView:    UIView?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Change this to locationView

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+    public var locationImage:        UIImageView?
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Change this to locationImageView

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+        subPlacesLabel.textColor = .darkGrayStant
+        subPlacesLabel.font      = .systemFont(ofSize: 12)
+        
+        let subPlaceText    = quantitySubPlaces == 1 ? "\(quantitySubPlaces) \(AppStrings.place_planning_sub_place_label)" :
+                              "\(quantitySubPlaces) \(AppStrings.place_planning_sub_places_label)"
+        subPlacesLabel.text = subPlaceText
+        
+        self.addSubview(subPlacesLabel)
+        subPlacesLabel.anchor(leading: locationImageView?.trailingAnchor,
+                              bottom: self.bottomAnchor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Indentation.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+        fispIconView.anchor(top:      self.topAnchor,
+                            trailing: self.trailingAnchor,
+                            padding:  UIEdgeInsets(top:    10,
+                                                   left:   0,
+                                                   bottom: 12,
+                                                   right:  8),
+                            size:     CGSize(width:  34,
+                                             height: 34))
+        
+        fispIconImage = UIImageView()
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Indentation.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+        guard let imageFispIcon = fispIconImage else { return }
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: siffIconImage

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+    fileprivate func configureGestureRecognizerLeftView(quantitySubPlaces: Int) {
+        if quantitySubPlaces == 0 { return }
+        
+        leftTapView                 = UIView()
+        guard let leftTapView       = leftTapView else { return }
+        leftTapView.backgroundColor = .clear
+        
+        self.addSubview(leftTapView)
+        leftTapView.anchor(top:  self.topAnchor,
+                        leading: self.leadingAnchor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Indent this parameters

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+        self.layer.applySketchShadow(color:  .shadowStant,
+                                     alpha:  0.09,
+                                     x:      0,
+                                     y:      3,
+                                     blur:   8,
+                                     spread: 3)
+    }
+}
+
+public protocol ListingPlacesCellDidSelectDelegate {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Change it to PlaceCellDidSelectDelegate

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    public var tableView:            UITableView?
+    public var placesStatusEnum:     [PlaceStatusEnum] = [.noPlanning, .hasPlanning, .allApproved, .allFinalized]
+    public var placeTitleList:       [String]          = ["Obra 1","Obra 2","Obra 3","Obra 4"]
+    public var quantitySubPlaceList: [Int]             = [0, 1, 2, 3]
+    public var percentageList:       [Float]           = [80, 32.5, 30, 110]
+    public var hasFispsList:         [Bool]            = [true, false, false, true]
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift:23]: Remove extra line.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+                         bottom:   self.view.bottomAnchor,
+                         trailing: self.view.trailingAnchor)
+        
+        tableView.register(ListingPlacesCell.self, forCellReuseIdentifier: ListingPlacesCell.identifier())
+    }
+}
+
+extension ListingPlacesViewController: UITableViewDataSource, UITableViewDelegate, ListingPlacesCellDidSelectDelegate {
+    public func goToSubPlaces() {
+        print("Have SubPlaces")
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift]: Has Subplaces. The subject is "it", so we use "has"

**Review by Leandro97 — COMMENTED** [review]
I realised that you aren't doing component tests. You need to do the tests. It doesn't need to be to complex, take ServiceInspectionFormFilledDetailMenuTests for example. There I test if the component initial state is alright and if the clicks are working properly.

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 1 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 2 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 3 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 4 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 5 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 9 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 10 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift (hunk 11 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cells/ListingPlacesCell/ListingPlacesCell.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift (hunk 12 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift:23]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift (hunk 13 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ListingPlaces/ListingPlacesViewController.swift]: Done

**Review by Leandro97 — APPROVED** [review]

**Review by andersonfernandes — APPROVED** [review]

**Pull request merged**